pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -299,6 +311,23 @@ public long getMaxTime() {
     return _maxTime;
   }
 
+  public void addExtraColumns(Schema newSchema) {
+    Map<String, BaseDefaultColumnHandler.DefaultColumnAction> defaultColumnActionMap =
```

Comment:
Not sure why DefaultColumnAction is needed here. We can just iterate through the schema and add a newly added column's FieldSpec if it is not there already. The FieldSpec can have the null value. There is no question of handling incompatible schema evolution at that point.